Automattic/wp-calypso
# Add all the Jetpack Atomic E2E Users & CI

Related to #

## Proposed Changes

This adds everything we need to start running Jetpack E2E tests against the special atomic sites during atomic Jetpack deploys.

To do this, this PR includes:

- [x] Adding all the users to the secrets.
- [x] Adding and supporting the new `ATOMIC_VARIATION`
- [x] Adding a Kotlin library function for making a test running step
- [x] Builds a new Atomic TeamCity build that loops through all the variations
- [x] Supporting the new `RUN_ID` and `RETRY_COUNT` env vars in suite and artifact titles
- [x] Updating existing e2e builds to use the `RETRY_COUNT` variable.

To control the scope and minimize regression testing, I only used the new Kotlin library function for the run tests step for the Jetpack deployment related builds.

## Testing Instructions

There's a few different pieces to test in different ways!

First, you can locally test the new ENV variable handling. With the right combinations, you should see the new Jetpack Atomic users in the run. You should also see updated test suite and artifact names. For example, consider the following local command...
`TEST_ON_ATOMIC=true JETPACK_TARGET=wpcom-deployment ATOMIC_VARIATION=php-old RUN_ID='Atomic: php-old' RETRY_COUNT=1  yarn jest specs/blocks/blocks__jetpack-earn.ts`
... you should see:
- The new Jetpack user for old PHP sites as the test user.
- A new suite name: `Blocks: Jetpack Earn (Atomic: php-old)`
- On failure, an artifact name that has both the `RUN_ID` and the `RETRY_COUNT` included!

Second, post-merge, there will be a new build for Jetpack Atomic deployment testing. It should run tests against all the different atomic variations. Again, can't test until merge.

Finally, all existing E2E builds should work as they did! Namely, the Jetpack Simple build should still run correctly, and all builds, when they have a retry, should use the retry correct in the artifact name. 

## Pre-merge Checklist
